src/main/webapp/app/shared/metis/posting-content/posting-content.components.ts

```diff
@@ -118,7 +118,7 @@ export class PostingContentComponent implements OnInit, OnChanges, OnDestroy {
                     referenceStr = this.content.substring(this.content.indexOf(']', patternMatch.startIndex)! + 1, this.content.indexOf('(', patternMatch.startIndex)!);
                     linkToReference = [this.content.substring(this.content.indexOf('(', patternMatch.startIndex)! + 1, this.content.indexOf(')', patternMatch.startIndex))];
                 } else if (ReferenceType.FAQ === referenceType) {
-                    referenceStr = this.content.substring(this.content.indexOf(']', patternMatch.startIndex)! + 1, this.content.indexOf('(', patternMatch.startIndex)!);
+                    referenceStr = this.content.substring(this.content.indexOf(']', patternMatch.startIndex)! + 1, this.content.lastIndexOf('(')!);
```

Comment:
I don't think you can use `context.lastIndexOf('(')` here. After all, there may be more parentheses after your FAQ reference that are unrelated to it.

For example, on TS3, if I write the message

`[faq]Question 1(/courses/31/faq?faqId=10)[/faq] test 1234 (`

I get the output

![image](https://github.com/user-attachments/assets/912ff3c8-baf6-485f-b60c-1634b8980aa7)

Other PRs (#9580, #9528) use a workaround `sanitizeStringForMarkdownEditor` where they remove problematic characters from the name of the referenced lecture/file. Maybe this could work instead of the change made here?